src/main/java/com/gravity9/mongocdc/MongoCDCManager.java

```diff
@@ -48,55 +49,78 @@ private Map<Integer, MongoChangeStreamWorker> createWorkers() {
-		workers.values().forEach(worker -> worker.register(listener));
-	}
+    /**
+     * Starts all workers for the specified collection in the MongoDB change data capture (CDC) manager.
+     * The workers will begin listening for change events and processing them accordingly.
+     * This method blocks until all workers are initialized and ready to process change events.
+     *
+     * @throws StartFailureException if an exception occurs during the start process
+     */
+    public void start() {
+        try {
+            log.info("Starting all workers for collection {}", clusterConfig.getCollection());
+            workers.values().forEach(MongoChangeStreamWorker::start);
+            workers.values().forEach(MongoChangeStreamWorker::awaitInitialization);
+            log.info("All workers for collection {} are now ready!", clusterConfig.getCollection());
+        } catch (Exception ex) {
+            try {
+                stop();
+            } catch (Exception exception2) {
+                log.error("Stop on exception failed", exception2);
+            }
+            throw StartFailureException.startFailure(ex);
+        }
+    }
+
+    /**
+     * Stops all workers for the specified collection in the MongoDB change data capture (CDC) manager.
+     * This method stops the workers from listening for change events and processing them.
+     * It also sets the thread reference to null, indicating that the worker has been stopped.
+     *
+     * @throws NullPointerException if the thread is null
+     */
+    public void stop() {
+        log.info("Stopping all workers for collection {}", clusterConfig.getCollection());
+        workers.values().forEach(MongoChangeStreamWorker::stop);
+        log.info("All workers for collection {} are now stopped!", clusterConfig.getCollection());
+    }
+
+    public void registerListener(ChangeStreamListener listener, Collection<Integer> partitions) {
+        partitions.forEach(partition -> {
+            var workerOptional = getMongoChangeStreamWorker(listener, partition);
+            workerOptional.ifPresent(worker -> worker.register(listener));
+        });
+    }
+
+    public void registerListenerToAllPartitions(ChangeStreamListener listener) {
+        workers.values().forEach(worker -> worker.register(listener));
+    }
+
+    public void deregisterListener(ChangeStreamListener listener, Collection<Integer> partitions) {
+        partitions.forEach(partition -> {
+            var workerOptional = getMongoChangeStreamWorker(listener, partition);
+            workerOptional.ifPresent(worker -> deregister(worker, listener, partition));
+        });
+    }
+
+    public void deregisterListenerFromAllPartitions(ChangeStreamListener listener) {
+        workers.values().forEach(worker -> worker.deregister(listener));
+    }
+
+    private void deregister(MongoChangeStreamWorker worker, ChangeStreamListener listener, int partition) {
```

Comment:
You can put all this code inside the worker - [Tell Don't Ask](https://martinfowler.com/bliki/TellDontAsk.html) and return `boolean` if needed (= `false` if listener not found).

Reply:
done. Return statement not added - i think it's not needed